### DISCUSSION TITLE
TagsInput relationship

### DISCUSSION THREAD

**Discussion by loranger** [filamentphp/filament#400] [Ideas]
Hello,

Is there a way to make tagsInput matching relationship ?
I have a model with polymorphic many to many relationship, and it would be very convenient to being able to attach detach relations using the TagInput feature, maybe with a TagInputBelongsTo component, maybe ?

**Comment by MACscr** [comment]
Yep, right now the tags design doesnt even make sense in my opinion as no one is going to store data comma separated like that.

**Reply by aurawindsurfing** [reply]
Hey @danharrin correct me if I'm wrong but currently the way to do this would be by using attach/detach methods on a Relationship manager while using ManytoMany polymorphic taggable table, there will be no way for us to use https://filamentphp.com/docs/2.x/forms/fields#tags-input unless in connection with https://github.com/spatie/laravel-tags

**Comment by aurawindsurfing** [comment]
Thanks!

**Comment by chrispage1** [comment]
I've created a proposed fix #14565

**Comment by kura1420** [comment]
```php
TagsInput::make('tags')
                    ->label('Tags')
                    ->placeholder('Add tags')
                    ->suggestions(Tag::pluck('name'))
                    ->trim()
                    ->afterStateHydrated(function (TagsInput $component, $state, $record) {
                        if ($record) {
                            $component->state(
                                $record->tags->pluck('tag.name')
                            );
                        }
                    })
                    ->columnSpanFull(),
```

you can use this
